addons/payment_razorpay/models/payment_transaction.py

```diff
@@ -405,8 +405,9 @@ def _process_notification_data(self, notification_data):
             if self.provider_id.capture_manually:
                 self._set_authorized()
         elif entity_status in const.PAYMENT_STATUS_MAPPING['done']:
-            if entity_data.get('token_id') and self.provider_id.allow_tokenization:
-                self._razorpay_tokenize_from_notification_data(notification_data)
+            if not self.token_id and entity_data.get('token_id') \
+               and self.provider_id.allow_tokenization:
```

Comment:
```suggestion
            if (
                not self.token_id
                and entity_data.get('token_id')
                and self.provider_id.allow_tokenization
            ):
```

Reply:
Done! Could you check the test case? :see_no_evil: